### PR TITLE
docs: simplify docs

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR tries to simplify the existing docs, moving pieces of text to the main docs, and removing outdated or value-less information that was there.

It basically keeps three main docs:

- README.md: contains the entrypoint for the information, setting the basis about e2e testing, BDD and tools
- .ci/README.md: contains the execution of the tests, so that it's possible to run the tests from local against AWS instances
- CONTRIBUTING.md: contains the minimal guide to start contributing to the tests, with a simple example

## Why is it important?
Remove texts that were causing more noise than value

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [x] I recommend reading this PR commit by commit, to understand each change

## How to test this PR locally
Reading those three docs: if they provide a good experience that you feel confortable to start working on the framework, then it's a success

### CODE REVIEW

File: .ci/ansible/tasks/setup_test_script.yml [hunk 1]
```diff
@@ -133,8 +133,8 @@
     - "GITHUB_CHECK_SHA1=\"{{ lookup('env', 'GITHUB_CHECK_SHA1') or '' }}\""
     - "GITHUB_CHECK_REPO=\"{{ lookup('env', 'GITHUB_CHECK_REPO') or 'elastic-agent' }}\""
     - "ELASTIC_APM_GLOBAL_LABELS=\"{{ lookup('env', 'ELASTIC_APM_GLOBAL_LABELS') }}\""
-    - "LOG_LEVEL=\"TRACE\""
-    - "OP_LOG_LEVEL=\"TRACE\""
+    - "LOG_LEVEL=\"{{ lookup('env', 'LOG_LEVEL') or 'TRACE' }}\""
```
Comment (mdelapenya) [inline .ci/ansible/tasks/setup_test_script.yml:136]: This change is not related to docs, but found it while testing the docs. Closed #1941

File: QUICKSTART.md [hunk 2]
```diff
@@ -1,10 +1,29 @@
 # Quickstart
 
-Let's walk through two quick examples to see how to start working with the e2e-testing framework. We have two similar use cases:
+Let's walk through two quick examples to see how to start working with the e2e-testing framework. But first, we need to understand how the tests work, and what dependencies we need to run them.
+
+- dependencies
+- how do the tests work?
+
+Once we understood that part, we can move on and continue with the code, where we have two similar use cases:
```
Comment (mdelapenya) [inline QUICKSTART.md:8]:
```suggestion
Once we understand that part, we can move on and continue with the code, where we have two similar use cases:
```

**Review by kuisathaverat — APPROVED** [review]

File: e2e/_suites/kubernetes-autodiscover/README.md [hunk 3]
```diff
@@ -100,24 +100,28 @@ feature files.
...
    # Depending on the versions used,
    export BEAT_VERSION=7.12.0 # version of beats to use
    export ELASTIC_AGENT_VERSION=7.12.0 # version of Elastic Agent to use
    export GITHUB_CHECK_SHA1=0123456789 # to select snapshots built by beats-ci
+   export KIND_VERSION="0.14.0" # version of kind
    export KUBERNETES_VERSION="1.24.0" # version of the cluster to be passed to kind
    ```
 
-3. Install dependencies.
+1. Install dependencies.
```
Comment (v1v) [inline e2e/_suites/kubernetes-autodiscover/README.md:117]:
```suggestion
3. Install dependencies.
```

File: e2e/_suites/kubernetes-autodiscover/README.md [hunk 4]
```diff
@@ -100,24 +100,28 @@ feature files.
...
 
    - Install Kubectl 1.18 or newer
    - Install Kind 0.14.0 or newer
    - Install Go, using the language version defined in the `.go-version` file at the root directory. We recommend using [GVM](https://github.com/andrewkroh/gvm), same as done in the CI, which will allow you to install multiple versions of Go, setting the Go environment in consequence: `eval "$(gvm 1.15.9)"`
    - Godog and other test-related binaries will be installed in their supported versions when the project is first built, thanks to Go modules and Go build system.
 
-4. Run the tests.
+2. Run the tests.
```
Comment (v1v) [inline e2e/_suites/kubernetes-autodiscover/README.md:124]:
```suggestion
4. Run the tests.
```

**Review by v1v — APPROVED** [review]
LGTM! Thanks for working on this :)

File: .ci/README.md [hunk 5]
```diff
@@ -116,6 +116,8 @@ In these VMs, the test framework will download a binary to install the Elastic A
 
 In order to list all the supported platforms, please run this command:
 
+> Docker is needed to run the command, as it internally uses `yq` in the form of a Docker container to read the YML file for the available platforms. To install and configure it, please read [this guide](https://docs.docker.com/engine/install/).
```
Comment (cachedout) [inline .ci/README.md:119]:
```suggestion
> Docker is needed to run the command, as it internally uses `yq` in the form of a Docker container to read the YML file for the available platforms. To install Docker, please read [this guide](https://docs.docker.com/engine/install/).
```

It is not clear here where `it` refers to Docker or to yq. I have modified the sentence to make it more explicit.

File: QUICKSTART.md [hunk 6]
```diff
@@ -1,10 +1,29 @@
 # Quickstart
 
-Let's walk through two quick examples to see how to start working with the e2e-testing framework. We have two similar use cases:
+Let's walk through two quick examples to see how to start working with the e2e-testing framework. But first, we need to understand how the tests work, and what dependencies we need to run them.
+
+- dependencies
```
Comment (cachedout) [inline QUICKSTART.md:5]: This is a bit odd. Was this perhaps a placeholder that you forgot to remove?

File: README.md [hunk 7]
```diff
@@ -19,7 +19,48 @@ This repository contains:
...
+
+## Statements about the test framework
+
+- It uses the `Given/When/Then` pattern to define scenarios.
+  - `Given` a state is present (optional)
+  - `When` an action happens (mandatory) 
+  - `Then` an outcome is expected (mandatory)
+  - `And/But` clauses are allowed to provide continuation on the above ones. (Optional)
+- Because it uses BDD, it's possible to combine existing scenario steps (the given-when-then clauses) forming new scenarios, so with no code it's possible to have new tests. For that reason, the steps must be atomic and decoupled (as any piece of software: low coupling and high cohesion).
+- It does not use the GUI at all, so there is no Selenium, Cypress or any other test library having expectations on graphical components. It uses Go as the programming language and Cucumber to create the glue code between the code and the plain English scenarios. Over the time, we have demonstrated that the APIs are not changing as fast as the GUI components.
```
Comment (cachedout) [inline README.md:58]:
```suggestion
- It does not use the GUI at all, so there is no Selenium, Cypress or any other test library having expectations on graphical components. It uses Go as the programming language and Cucumber to create the glue code between the code and the plain English scenarios. Over time, we have demonstrated that the APIs are not changing as fast as the GUI components.
```

File: QUICKSTART.md (hunk 6 above)
Comment (mdelapenya) [reply QUICKSTART.md:5]: Let me rephrase it:

```
First, we need to understand how the tests work, and what dependencies we need to run them.

- dependencies
- how do the tests work?
```
we add those two sections. And after that:

```
## Adding a new test suite

Let's walk through two quick examples to see how to start working with the e2e-testing framework.
```

Comment (mdelapenya) [reply QUICKSTART.md]: @cachedout I changed it in https://github.com/elastic/e2e-testing/pull/2838/commits/5d6668cb85e804b3a24d63850df66034d791ee95

**Review by cachedout — APPROVED** [review]
